inference-engine/thirdparty/clDNN/kernel_selector/core/kernel_selector.cpp

```diff
@@ -118,6 +121,9 @@ KernelsData kernel_selector_base::GetAutoTuneBestKernel(const Params& params,
     KernelsData kernelsData;
     std::string kernelName;
 
+    optional_params* p_options = const_cast<optional_params *>(&options);
```

Comment:
This part looks like a dirty hack. Can we have a better solution (i.e. keep unique_id in `params` structure only?)

Reply:
There are three options to pass unique_id to the GetEntryPoint().
1. set Params.UniqueID using program.id and program_node.unqiue_id and pass Params.UniqueID to optional_params.UniqueID
(program_node.unique_id -> params.unique_id -> optional_params.unique_id -> KernelID)

> - I use this way currently.
> - no interface change.

2. set optional_params.unique_id without Params.UniqueID
(program_node.unique_id -> optional_params.unique_id -> KernelID)

> - Need to change parameters of get_default_optional_params from program to program_node. it impacts 53 primitive_impl creation functions.

3. set params.uniqueID and generate KernelID without optional_params.UniqueID
(program_node.unique_id -> params.unique_id -> KernelID)

> - Need to add Params parameter of KernelBaseOpenCL::GetEntryPoint . it impacts GetCommonKernelsData in 73 KernelBases.


I've concerned that extensive code changes will make frequent code conflict and side effects by mistake. 
so I choose first option. but by your comments, I change the code to use mutable variable instead of const_cast.

If it is okay to change interface, it is available to use option 2 or 3. thanks.

Reply:
@vladimir-paramuzov, @amolotko, @sshlyapn I made new trial PR to change interface of set_optional_params by @amolotko 's comment(Option 2). Could you please review this PR too? 
https://github.com/ahnyoung-paul/openvino/pull/17 
- new PR use only optional_params.UniqueID

If this approach is OK, I will merge the PR to this parallel_choose_impl. thanks.

Reply:
@vladimir-paramuzov  Could you review the change of https://github.com/ahnyoung-paul/openvino/pull/17 ? If there is no problem we will apply it to this PR.

Reply:
@vladimir-paramuzov I merged the PR to keep unique id in Params. please review this PR again. thanks.